### PR TITLE
Match + link J2DGrafContext.cpp

### PR DESCRIPTION
**Pull request by HeartPiece44**
Matches and links J2DGrafContext.cpp, largely scaffolded using the TP decomp version as a guide (thank you, legends). Also fixes associated headers, including a repo-wide rework of `JUtility::TColor` to inherit from GXColor.

### CODE REVIEW

File: include/Dolphin/gx.h [hunk 1]
```diff
@@ -138,6 +138,16 @@ typedef enum _GXIndTexWrap {
 	GX_IND_TEX_WRAP_0,
 } GXIndTexWrap;
 
+typedef enum _GXTexOffset {
+	/* 0x0 */ GX_TO_ZERO,
+	/* 0x1 */ GX_TO_SIXTEENTH,
+	/* 0x2 */ GX_TO_EIGHTH,
```
Comment (jonathonbrady) [inline include/Dolphin/gx.h:144]: is this also from TP? are we using this comment convention anywhere else?

File: src/plugProjectEbisawaU/efx2dBase.cpp [hunk 2]
```diff
@@ -134,7 +134,7 @@ void TForever::setGlobalEnvColor(JUtility::TColor& color)
 	JPABaseEmitter* emitter = m_emitter;
 
 	if (emitter != nullptr) {
-		setColor(m_emitter->m_color, color);
+        m_emitter->m_color.setRGB(color);
```
Comment (jonathonbrady) [inline src/plugProjectEbisawaU/efx2dBase.cpp:137]: formatting

File: include/JSystem/JUT/TColor.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	void set(u32 u32Color) { *(u32*)&r = u32Color; }
+
+	void set(GXColor gxColor) { *(GXColor*)&r = gxColor; }
+
+    void setRGB(u8 cR, u8 cG, u8 cB) {
+        r = cR;
+        g = cG;
+        b = cB;
```
Comment (jonathonbrady) [inline include/JSystem/JUT/TColor.h:43]: formatting

File: src/JSystem/J2D/J2DGrafContext.cpp [hunk 4]
```diff
@@ -280,604 +96,176 @@ void J2DGrafContext::setup2D()
...
+	m_colorBL = colorBL;
+
+	_B0        = J2DBlendInfo(1, 4, 5);
+	m_linePart = J2DBlendInfo(1, 4, 5);
+	m_boxPart  = J2DBlendInfo(1, 4, 5);
+
+	if ((u8)u32(m_colorTL) != 0xFF) {
+		return;
+	}
+
```
Comment (jonathonbrady) [inline src/JSystem/J2D/J2DGrafContext.cpp:153]: thoughts on a u8 operator?

**Review by jonathonbrady — CHANGES_REQUESTED** [review]
looks good overall, just a few nitpicks to address before merging

File: include/Dolphin/gx.h (hunk 1 above)
Comment (HeartPiece44) [reply include/Dolphin/gx.h]: Yep, these were from TP - forgot to integrate our comment style with them, but fixed it! ty for spotting it!

File: src/JSystem/J2D/J2DGrafContext.cpp (hunk 4 above)
Comment (HeartPiece44) [reply src/JSystem/J2D/J2DGrafContext.cpp:153]: I haven't seen this crop up yet outside of this one singular instance, so I'm not sure how useful/warranted it'd be. I think if it comes up again it's probably worth working in a u8 operator though? Worth keeping an eye on.

**Pull request merged**